### ISSUE TITLE
Feature: Check if namespacing class does not inherit at the same time

### ISSUE THREAD

**Issue by ipepe** [rubocop-hq/rubocop#7441]
Today I saw a code similar to this (note: class User is defined in other file):
```ruby
class User < ApplicationRecord
  module Something
    class SomeService
      # code here
    end
  end
end
```

What it should look like instead:

```ruby
class User
  module Something
    class SomeService
      # code here
    end
  end
end
```

I would like a Cop to mark first snipped construction as bad

**Comment by buehmann** [comment]
Hi @ipepe, can you give us some details on why you consider the first snipped bad? I think we are lacking some context here that might be clear for you.

**Comment by ipepe** [comment]
Consider following application code:

Step 1.
`app/models/user.rb`:
```ruby
class User < ApplicationRecord
  # validates something
  # some authorization
end
```

`app/services/user/something/some_service.rb`:
```ruby
class User < ApplicationRecord
  module Something
    class SomeService
      # code here
    end
  end
end
```

Step 2.
You want to change inheritance in user class:
`app/models/user.rb`:
```ruby
class User < AuthorizedResource
  # validates something
end
```

Step 3. 

What happens when Your service file reopens User class with different inheritance?
